### ISSUE TITLE
react/boolean-prop-naming interface/type on React.FC validation of boolean with typescript does not work

### ISSUE THREAD

**Issue by daphnesmit** [yannickcr/eslint-plugin-react#3197]
So this works:
```
type Props = {
  enabled: boolean
}
const Hello = (props: Props) => <div />;
```

But If you want to validate a boolaan prop on an interface or type in a React.FC
```

type Props = {
  enabled: boolean;
}

const Hello: React.FC<Props> = ({ enabled }) => {
```

Then it doesnt work and thinks enabled is allowed.
Is this intended?

**Comment by ithater** [comment]
It seems it does not work with interfaces either.

This code is not complaining
```js
interface Props {
  enabled: boolean;
}

const Hello = (props: Props) => <div />;
```

**Comment by hollandThomas** [comment]
I _think_ this occurs when using [the new JSX transform](https://reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html) which does not require you to import `React` when using JSX.

```tsx
// this does not work
type Props = { enabled: boolean };
const Hello: React.FC<Props> = ({ enabled }) => null
```

```tsx
// this works
import type { FC } from 'react';

type Props = { enabled: boolean };
const Hello: FC<Props> = ({ enabled }) => null
```
